contrib/mesos/pkg/minion/server.go

```diff
@@ -164,7 +164,7 @@ func (ms *MinionServer) launchProxyServer() {
 // executor doesn't support failover right now, the right thing to do is to fail completely since all
 // pods will be lost upon restart and we want mesos to recover the resources from them.
 func (ms *MinionServer) launchExecutorServer(containerID string) <-chan struct{} {
-	allArgs := os.Args[1:]
+	allArgs := os.Args[2:]
```

Comment:
i want to clarify my previous comment: you're not required to run "km
minion" in a custom executor script, you're welcome to spin up the
kubelet-executor and kube-proxy in some custom script and totally ignore
the minion glue. but the minion code is where the first 2 args are stripped
from the command line, and the only way to run the minion code is via "km
minion"

On Fri, Mar 4, 2016 at 2:28 PM, James DeFelice <james@mesosphere.io> wrote:

> thanks for reviewing this change. the minion is only executable via "km
> minion" right now, there is no separate binary. so even if you were
> wrapping with a custom script and specifying an executor-path, you'd still
> need to invoke "km minion ..args" which means that the first two command
> line args may be safely stripped. if we ever support running the "km
> minion" service outside of the km binary then .. yes this assumption
> breaks. however, we have no immediate plans to support that.
>
> does that make sense, or am I still overlooking something?
>
>
> On Fri, Mar 4, 2016 at 1:59 PM, ravilr <notifications@github.com> wrote:
>
>> In contrib/mesos/pkg/minion/server.go
>> <https://github.com/kubernetes/kubernetes/pull/22500#discussion_r55072764>
>> :
>>
>> > @@ -164,7 +164,7 @@ func (ms *MinionServer) launchProxyServer() {
>> >  // executor doesn't support failover right now, the right thing to do is to fail completely since all
>> >  // pods will be lost upon restart and we want mesos to recover the resources from them.
>> >  func (ms *MinionServer) launchExecutorServer(containerID string) <-chan struct{} {
>> > -	allArgs := os.Args[1:]
>> > +	allArgs := os.Args[2:]
>>
>> @jdef <https://github.com/jdef>, looks like there is still a case where
>> arg parsing would break: if --executor-path flag set on km scheduler:
>>
>> https://github.com/kubernetes/kubernetes/blob/679e27c157637290075e65a45dfc0fde0a5c32d9/contrib/mesos/pkg/scheduler/service/service.go#L385
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kubernetes/kubernetes/pull/22500/files#r55072764>.
>>
>
>